.github/workflows/linkcheck.yml

```
@@ -27,18 +29,22 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
       with:
-        fetch-depth: 0
+        persist-credentials: false
 
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: 3.12
+    - name: Install Graphviz
+      uses: ts-graphviz/setup-graphviz@v2
+
+    - name: Install pandoc
+      # This step requires Ubuntu
+      run: |
+        sudo apt update
+        sudo apt-get install -y pandoc
 
-    - name: Install Python dependencies
-      run: python -m pip install --upgrade nox uv
+    - name: Install uv
+      uses: astral-sh/setup-uv@v5
```

```suggestion
      uses: astral-sh/setup-uv@v5
      with:
        python-version: 3.12
```

Then I do not think you need to use `--python 3.12` in the `uv` command on line 50.